nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/LoomServer.java

```diff
@@ -90,13 +92,14 @@ class LoomServer implements WebServer {
                                              socketName,
                                              socketConfig,
                                              router,
-                                             simpleHandlers));
+                                             simpleHandlers,
+                                             inheritThreadLocals));
         }
 
         this.listeners = Map.copyOf(listeners);
         this.executorService = Executors.newThreadPerTaskExecutor(Thread.ofVirtual()
                                                                           .allowSetThreadLocals(true)
-                                                                          .inheritInheritableThreadLocals(false)
+                                                                          .inheritInheritableThreadLocals(true)
```

Comment:
seems you hardcoded this to true, you should use the variable `inheritThreadLocals`